### PR TITLE
Add webapi basic-auth

### LINKED ISSUE #783: Add basic-auth
## Description

Add [basic-auth](https://echo.labstack.com/middleware/basic-auth) as a replacement of the previous JWT

### PR DESCRIPTION
**Pull request by capossele**
This PR adds support to http basic-auth for:

- web-api
- client lib
- cli-wallet

Fixes #783

### CODE REVIEW

File: client/lib.go [hunk 1]
```diff
@@ -30,18 +30,82 @@ const (
...
+// NewGoShimmerAPI returns a new *GoShimmerAPI with the given baseURL and options.
+func NewGoShimmerAPI(baseURL string, setters ...Option) *GoShimmerAPI {
+	args := &Options{}
+
+	for _, setter := range setters {
+		setter(args)
+	}
+	if args.HTTPClient == nil {
+		args.HTTPClient = &http.Client{}
+	}
```
Comment (jkrvivian) [inline client/lib.go:96]: Don't need this check if we have `newOptions`.
```
func newOptions() *Options {
    return &Options{
         BasicAuth: &BasicAuth{},
         HTTPClient: &http.Client{},
    }
}
```

Comment (capossele) [reply client/lib.go]: @jkrvivian good suggestions! I've refactored the way we handle options, so that now feels more intuitive (I hope)!

Comment (jkrvivian) [reply client/lib.go]: That's even better! :smile:

**Review by jkrvivian — APPROVED** [review]

File: plugins/webapi/plugin.go [hunk 2]
```diff
@@ -47,6 +47,17 @@ func Server() *echo.Echo {
...
+			server.Use(middleware.BasicAuth(func(username, password string, c echo.Context) (bool, error) {
+				if username == config.Node().GetString(CfgBasicAuthUsername) &&
+					password == config.Node().GetString(CfgBasicAuthPassword) {
+					return true, nil
+				}
+				return false, nil
+			}))
+		}
 	})
 	return server
```
Comment (Wollac) [inline plugins/webapi/plugin.go:62]: The error server does not have an `HTTPErrorHandler` set, this should be done to log errors similar to `plugin/dashboard`

**Review by Wollac — CHANGES_REQUESTED** [review]
I also have some remarks about stuff not actually touched in the PR:

Unauthorized access is not logged correctly:
In https://github.com/iotaledger/goshimmer/blob/fd03b22ff7b4ddde48a4907be55455c45bbfb704/plugins/dashboard/routes.go#L77 
this should be replaced with 
```log.Warnf("Request failed: %s", err)```
to log errors in our usual style.

In https://github.com/iotaledger/goshimmer/blob/fd03b22ff7b4ddde48a4907be55455c45bbfb704/plugins/webapi/plugin.go#L87
it could make sense to also log whether basic_auth is on or not.

Same for https://github.com/iotaledger/goshimmer/blob/fd03b22ff7b4ddde48a4907be55455c45bbfb704/plugins/dashboard/plugin.go#L115

**Review by jkrvivian — APPROVED** [review]

**Review by jkrvivian — APPROVED** [review]

**Comment by frohal** [comment]
A small hint on how to create the password-hash to put in the JSON  could possibly have been added as well ?
@Wollac

**Pull request merged**